File: leetcode/00014. Longest Common Prefix/Python/sol.py
# Bruteforce Solution
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        out = strs.pop()
        for i in strs:
            l1 = len(out)
            l2 = len(i)
            j = 0
            str = ""
            while j < l1 and j < l2:
                if out[j] == i[j]:
                    str += out[j]
                    j+=1
                else:
                    break
            out = str
        return out

File: leetcode/00014. Longest Common Prefix/Rust/sol.rs
impl Solution {
    pub fn longest_common_prefix(mut strs: Vec<String>) -> String {
        let mut prefix = strs[0].clone();
        for s in strs.iter().skip(1) {
            while !s.starts_with(&prefix) {
                prefix.pop();
                if prefix.is_empty() {
                    return String::new();
                }
            }
        }
        return prefix;
    }
}